### PR TITLE
fix: do not hydrate demo page

### PR DESCRIPTION
**Pull request by MadCcc**
### 🤔 这个变动的性质是？/ What is the nature of this change?

- [ ] 新特性提交 / New feature
- [x] bug 修复 / Fix bug
- [ ] 样式优化 / Style optimization
- [ ] 代码风格优化 / Code style optimization
- [ ] 性能优化 / Performance optimization
- [ ] 构建优化 / Build optimization
- [ ] 网站、文档、Demo 改进 / Website, documentation, demo improvements
- [ ] 重构代码或样式 / Refactor code or style
- [ ] 测试相关 / Test related
- [ ] 其他 / Other

### 📝 更新日志 / Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix that demo page should not be hydrated.        |
| 🇨🇳 Chinese |   修复 demo 页会被注水的问题。        |

### CODE REVIEW

File: src/features/exportStatic.ts [hunk 1]
```diff
@@ -50,4 +47,25 @@ export default (api: IApi) => {
       return memo;
     },
   });
+
+  api.onGenerateFiles(() => {
+    api.writeTmpFile({
+      path: 'core/exportStaticRuntimePlugin.ts',
```
Comment (PeachScript) [inline src/features/exportStatic.ts:53]: dumi 的插件统一写到 dumi 下哈
```suggestion
      path: 'dumi/exportStaticRuntimePlugin.ts',
```

File: src/features/exportStatic.ts [hunk 2]
```diff
@@ -50,4 +47,25 @@ export default (api: IApi) => {
...
+      content: `
+export function modifyClientRenderOpts(memo: any) {
+  const { history, hydrate } = memo;
+
+  return {
+    ...memo,
+    hydrate: hydrate && !history.location.pathname.startsWith('/~demo'),
```
Comment (PeachScript) [inline src/features/exportStatic.ts:60]: ~ 用 `SP_ROUTE_PREFIX` 代替？demo 应该是 `demos`

Comment (MadCcc) [reply src/features/exportStatic.ts]: done

File: src/features/exportStatic.ts (hunk 1 above)
Comment (MadCcc) [reply src/features/exportStatic.ts]: done

**Review by PeachScript — APPROVED** [review]
+1

**Pull request merged**